astropy/io/fits/tests/test_image.py

```diff
@@ -1790,6 +1790,33 @@ def test_compressed_integers(self, dtype):
         new = fits.getdata(testfile)
         np.testing.assert_array_equal(data, new)
 
+    def test_write_non_contiguous_data(self):
+        """
+        Regression test for https://github.com/astropy/astropy/issues/2150
+        """
+        orig = np.arange(100, dtype=float).reshape((10, 10), order='f')
+        assert not orig.flags.contiguous
+        primary = fits.PrimaryHDU()
+        hdu = fits.CompImageHDU(orig)
+        hdulist = fits.HDUList([primary, hdu])
+        hdulist.writeto(self.temp('test.fits'))
+
+        actual = fits.getdata(self.temp('test.fits'))
+        assert_equal(orig, actual)
```

Comment:
I wonder if exact match required for floating point comparison will come back to bite us, but let's leave it as-is for now.
```suggestion
        assert_equal(orig, actual)
```